core/src/main/scala/org/genivi/sota/core/DeviceUpdatesResource.scala

```diff
@@ -115,11 +116,23 @@ class DeviceUpdatesResource(db: Database,
   /**
     * An ota client GET the binary file for the package that the given [[UpdateRequest]] and [[Device]] refers to.
     */
-  def downloadPackage(device: Uuid, updateId: Refined[String, Uuid.Valid]): Route = {
+  def downloadPackage(device: Uuid, updateId: Refined[String, Uuid.Valid]): Route =
     withRangeSupport {
-      complete(packageDownloadProcess.buildClientDownloadResponse(device, updateId))
+      val download = packageDownloadProcess.buildClientDownloadResponse(device, updateId)
+
+      if (userProfile.isDefined) {
```

Comment:
I'm not super happy about the pattern if (XXX.isDefined) { ... XXX.get ...} Fred McBride did not invent pattern-matching for nothing you know :)